### DISCUSSION TITLE
No further readings from G7 on Pixel 7pro

### DISCUSSION THREAD

**Discussion by Nawygator** [NightscoutFoundation/xDrip#3611] [Q&A]
Hey there,
I am using a Google Pixel 7 Pro on Android 15 beta, Dexcom G7, xdrip nightly.
It worked until the last Android update. Now, I only get the actual reading (and past values) directly after pairing, but afterwards I only get power connect errors, Auth errors and similar.
Workaround: Right now I change the "dexcom sender id" to a different one and back to the original, delete the BT device and then restart my phone. After 5 min I can Accept the pairing request and get the past values.
I tried changing most of the options for dexcom and bt, but nothing could help me out of this loop.
TIA, im grateful for any tips.
![Screenshot_20240804-165510.png](https://github.com/user-attachments/assets/d76a8788-eea6-4ac4-8644-56f998942957)

![Screenshot_20240804-165520.png](https://github.com/user-attachments/assets/920bdf54-0e00-4fdc-935f-6494f1d3fcea)

![Screenshot_20240804-170530.png](https://github.com/user-attachments/assets/f9d44af1-497c-49ca-ac98-933903c20e8a)

![Screenshot_20240804-170537.png](https://github.com/user-attachments/assets/200a1073-c1f2-42b0-96f3-28c486874ef4)

**Comment by Navid200** [comment]
How old is that G7?
Can you check the logs to see how fast Voltage B has been dropping?  
https://navid200.github.io/xDrip/docs/Dexcom/BatteryVoltageLogs.html

**Reply by Nawygator** [reply]
This G7 is about 3 days old, but I had the same problem with the last one as well. I thought the problem might have been the last sensor, but now I have a new sensor and the same problem.
The problem with low voltage is new, before low voltage I already got no readings

**Reply by Navid200** [reply]
Do you mind looking at the logs anyway?

**Reply by Nawygator** [reply]
![Screenshot_20240804-181346.png](https://github.com/user-attachments/assets/c4270f76-0269-4faf-9c27-206bbf15b801)

**Reply by Navid200** [reply]
How long ago did you receive these sensors?  Could it have been more a year ago?  

Have you disabled xDrip battery optimization?  Have you set it to unrestricted?  
Are all xDrip settings in agreement with these?  https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

**Reply by Nawygator** [reply]
I received the sensor about a month ago. 
I changed most options in the last few days, but I just changed them back to your recommendations. I also checked the battery optimization regarding xdrip and native Bluetooth. It seems to me that the update restricted Bluetooth . I restarted my phone today countless times, even after recovering your recommendations and setting battery optimization for Bluetooth to be unrestricted. 
I got one reading, but afterwards deep sleeping errors again.

**Reply by Navid200** [reply]
What do you do with your old used G7 devices?  Are they sitting somewhere nearby in a drawer?  

I'm sorry I understood from your message that you disabled Bluetooth optimization.
But, what about xDrip battery optimization?  Did you set that to unrestricted also?

**Reply by Nawygator** [reply]
There are no old devices near me.
Xdrip battery optimization was already disabled. 
It seems to work since I set Bluetooth battery optimization to unrestricted.
I think, that the android update changed the setting.
Thank you very much.

**Reply by Navid200** [reply]
You're welcome

**Reply by Nawygator** [reply]
Hey,
Yesterday my sensor failed on day 8, but worked since we wrote.
The new one acts in the same old way, that I only get one reading. 
I checked all options again. They are all just like the recommendations, including battery optimization.
I looked into the log, found this one a few times.

RxJavaError2:io.reactivex.exceptions. On ErrorNotimplemented Exception The exception was not handled due to missing onError handler in the subscribe) method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | com.polidea.rxandroidble2.exceptions.BleGattCharacteristic Exception: GATT exception from MAC='XX:XX:XX:XX:XX:XX, status 133 (GATT_ERROR), type BleGattOperation(description='CHARACTERISTIC_WRITE). (Look up status 0x85 here https://android.googlesource.com/platform/external/bluetooth/bluedroid/+/android-5.1.0_r1/stack/include/gatt _api. h) Functions$OnErrorMissing Consumer::accept -> Functions$OnErrorMissing Consumer:accept

May this helps?

**Reply by Navid200** [reply]
What does the status page show?

**Reply by Nawygator** [reply]
![Screenshot_20240810-100834.png](https://github.com/user-attachments/assets/6c72a633-0f7b-439d-94a6-7a1e4abd0ada)

And every 5 min there is this in the log: 

Failed to write SensorTxMessage: com.polidea.rxandroidble2.exceptions. BleGattCharacteristic Exception: GATT exception from MAC='XX:XX:XX:XX:XX:XX, status 133 (GATT_ERROR), type BleGattOperation{description=CHARACTERISTIC_WRITE}. (Look up status Ox85 here https://android.googlesource. com/platform/external/bluetooth /bluedroid/+/android-5.1.0r1 /stack/include/gatt api.h)

**Reply by Navid200** [reply]
When did you receive this G7?
Has it been on a shelf for a year or something like that?
Do you have the box?  You can see the date it was manufactured on it.

**Reply by Nawygator** [reply]
It was manufactured in April of this year

**Reply by Navid200** [reply]
My Pixel 6a misbehaved when Bluetooth battery optimization had automatically enabled after an Android patch.   After I set it to unrestricted again, it has been reliable.

**Reply by Nawygator** [reply]
I think, that this happened last week to me. It has worked for 5 days since then. There was no update in the meantime. Bluetooth is unrestricted, but I have many missing measurements at the moment again.

**Reply by Navid200** [reply]
One thing that surprises me is how low voltage B is.  Are you sure these are not very old devices?

**Reply by Nawygator** [reply]
I am absolutely sure, they were produced in April 2024.

**Reply by Navid200** [reply]
It is very interesting and puzzling.
If you look at the image at the bottom of the following linked page, you will see the voltages I have experienced with 14 different devices.  They agree with most readings I have also seen reported by others.
https://navid200.github.io/xDrip/docs/Battery-condition.html

**Reply by Nawygator** [reply]
Voltage B is after 2,5 Days currently at 215.
 It seems to me, I got a bad batch. Every Sensor is produced on the 1st of April 2024.

**Reply by Nawygator** [reply]
Sometimes it works for a few hours, sometimes just one reading.
I need to change the sensor id and delete it from Bluetooth, but at least I don't have to restart my phone 15 times a day...
I have been using G6 and G7 since 2016, but the last weeks are the worst

**Reply by Nawygator** [reply]
Sometimes it works for a few hours, sometimes just one reading.
I need to change the sensor id and delete it from Bluetooth, but at least I don't have to restart my phone 15 times a day...
I have been using G6 and G7 since 2016, but the last weeks are the worst

**Reply by Navid200** [reply]
What happens if you use Dexcom app?  Does it behave the same way?  Or, does it work fine with Dexcom app?  That will be a crucial piece of info.

You can just switch xDrip to disabled collector:  https://navid200.github.io/xDrip/docs/Stop-xDrip.html
Then, you won't have to uninstall xDrip.  Then, install Dexcom app and follow the instruction to connect to the current G7.  Does it work?

**Comment by ddellspe** [comment]
I'm running into this same issue, not sure if there was a resolution or not, but it's connecting just fine to my Tandem Insulin Pump, not getting connection to my phone at all

```
2024-09-04 17:51:59 ^ Ob1G5CollectionService ^ Error count reached: 2
2024-09-04 17:51:59 ^ Ob1G5CollectionService ^ RxJavaError2:io.reactivex.exceptions.OnErrorNotImplementedException The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | com.polidea.rxandroidble2.exceptions.BleGattCharacteristicException: GATT exception from MAC='XX:XX:XX:XX:XX:XX', status 133 (GATT_ERROR), type BleGattOperation{description='CHARACTERISTIC_WRITE'}. (Look up status 0x85 here https://android.googlesource.com/platform/external/bluetooth/bluedroid/+/android-5.1.0_r1/stack/include/gatt_api.h) Functions$OnErrorMissingConsumer::accept -> Functions$OnErrorMissingConsumer::accept
```
Also seeing the following in the log:
```
2024-09-04 17:37:20 ^ Ob1G5StateMachine ^ Got exception in plugin: java.lang.SecurityException: Mismatch - wait
```

Brand new sensor, unable to even get any battery information

**Reply by ddellspe** [reply]
I've really been struggling with even getting the Dexcom G7 app to pair with this sensor, I kind of assume this is just a garbage sensor, was able to connect to my insulin pump without issues, but phone has been a nightmare (had pairing issue with PIN, etc. but was finally able to get the G7 app connected

**Reply by jamorham** [reply]
Move old removed sensors far away as they can interfere which is what the mismatch errors are likely about.

**Reply by ddellspe** [reply]
it took 8 hours, and I had to turn on OB1 unbonding at least momentarily, but I was able to get it started finally at 3:00 AM when I was somewhat having insomnia.  I think this sensor just has loads of different problems with it, I had it fail with the same scenario during runtime on the G7 app directly too, but it was consistently working on my insulin pump.  Still not sure how I feel about this generation of sensors

**Comment by nickb24** [comment]
I've been debugging a similar issue on my end for the past week. This looks like it might be similar, but I can't tell unless someone is able to post a logcat log (ideally) or at the very least logs after enabling this: https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector
